src/AutoRest.CSharp/Common/Input/InputTypes/Serialization/TypeSpecReferenceHandler.cs

```diff
@@ -24,9 +24,9 @@ private class TypeSpecReferenceResolver : ReferenceResolver
 
             public override void AddReference(string referenceId, object value)
             {
-                if (!_referenceIdToObjectMap.TryAdd(referenceId, value))
+                if (!_referenceIdToObjectMap.ContainsKey(referenceId))
                 {
-                    throw new JsonException();
+                    _referenceIdToObjectMap.Add(referenceId, value);
```

Comment:
this scenario "adding a reference when the same Id existed" should never happen, therefore I believe we should revert this change.
We should make sure that definition of a model should only happen once.

Reply:
No. we cannot make the model.derievedModels only once otherwise we will meet circle loop issue of the json deserialization issue.

Reply:
Let us add a TODO here to emphasize this is a workaround.

The solution of this issue should be in the emitter to ensure it will not output the derived type twice and we should NEVER get a duplicated key here